sky/clouds/aws.py

```diff
@@ -114,27 +119,51 @@ def get_default_ami(cls, region_name: str, instance_type: str) -> str:
-        if image_id is not None:
-            if None in image_id:
-                image_id = image_id[None]
-            else:
-                assert region_name in image_id, image_id
-                image_id = image_id[region_name]
-            if image_id.startswith('skypilot:'):
-                image_id = service_catalog.get_image_id_from_tag(image_id,
-                                                                 region_name,
-                                                                 clouds='aws')
-                if image_id is None:
-                    # Raise ResourcesUnavailableError to make sure the failover
-                    # in CloudVMRayBackend will be correctly triggered.
-                    # TODO(zhwu): This is a information leakage to the cloud
-                    # implementor, we need to find a better way to handle this.
-                    raise exceptions.ResourcesUnavailableError(
-                        f'No image found for region {region_name}')
-            return image_id
-        return cls.get_default_ami(region_name, instance_type)
+    def _get_image_id(
+        cls,
+        image_id: Optional[Dict[str, str]],
+        region_name: str,
+    ) -> str:
+        if image_id is None:
+            return None
+        if None in image_id:
+            image_id = image_id[None]
+        else:
+            assert region_name in image_id, image_id
+            image_id = image_id[region_name]
+        if image_id.startswith('skypilot:'):
+            image_id = service_catalog.get_image_id_from_tag(image_id,
+                                                             region_name,
+                                                             clouds='aws')
+            if image_id is None:
+                # Raise ResourcesUnavailableError to make sure the failover
+                # in CloudVMRayBackend will be correctly triggered.
+                # TODO(zhwu): This is a information leakage to the cloud
+                # implementor, we need to find a better way to handle this.
+                raise exceptions.ResourcesUnavailableError(
+                    f'No image found for region {region_name}')
+        return image_id
+
+    def get_image_size(self, image_id: str, region: Optional[str]) -> float:
+        if image_id.startswith('skypilot:'):
+            return DEFAULT_AMI_GB
+        assert region is not None, (image_id, region)
+        client = aws.client('ec2', region_name=region)
+        try:
+            image_info = client.describe_images(ImageIds=[image_id])
+            image_info = image_info['Images'][0]
+            image_size = image_info['BlockDeviceMappings'][0]['Ebs'][
+                'VolumeSize']
+        except aws.botocore.exceptions.NoCredentialsError:
+            # Fallback to default image size if no credentials are available.
+            # The credentials issue will be handled when actually provisioning
+            # the instance.
```

Comment:
By handled, do you mean it will be caught and an appropriate error will be generated at provisioning time? If so, it would be nice to explicitly state that here: 
```
# Fallback to default image size if no credentials are available.
# The credentials issue will be caught when actually provisioning
# the instance and appropriate errors will be raised there.
```
(Mentioning because I thought the provisioning pipeline has some way to resolve this exception)

Reply:
Good point! I think it is the case. Replaced with the suggested comment. Thanks!